### ISSUE TITLE
[bug] Backslash in UnnecessaryGString causes Triple Double-Quote String to incorrectly be fixed to Triple Single-Quote breaking interpolation

### ISSUE THREAD

**Issue by darthwalsh** [nvuillam/npm-groovy-lint#157]
Starting code

```groovy
def path = "C:\\b"

def a = 'a'

def b = """
  ${a}
"""
```

Run `npm-groovy-lint --fix`

### Actual:
```groovy
def path = "C:\\b"

def a = 'a'

def b = '''
  ${a}
'''
```

The string has changed to single-quote, breaking interpolation.

### Expected
```groovy
def path = 'C:\\b'

def a = 'a'

def b = """
  ${a}
"""
```

`path` should be converted to single quotes, not `b`.

---

I originally created bug https://github.com/CodeNarc/CodeNarc/issues/597 but they showed that CodeNarc was reporting the correct rule violation; the `--fix` logic in npm-groovy-lint was causing the wrong behavior.

**Comment by zlin179** [comment]
Any update on this?

**Comment by darthwalsh** [comment]
@zlin179  based on the lack of activity it still seems broken
